### PR TITLE
fix bug with multy dot or dot in start of line

### PR DESCRIPTION
**Pull request by yurybotov**
Fix bug with sequence of dot or dot in start of line. 
The error was discovered by Max while testing the XOD library. I had to add an extra fixed-length buffer as a result the maximum length of a single text or digital field is now limited to 32 bytes. Do you think it is necessary to indicate this in the API? A real user will almost never work with such long chains of modules.

### CODE REVIEW

File: API_ru.md [hunk 1]
```diff
@@ -45,7 +45,7 @@
 
 ### `void display(String string, uint8_t position, uint8_t width, uint8_t flags = SEGM8_ALIGN_LEFT)`
 
-Записывает число или текстовую строку, начиная с позиции `position`, шириной `width` знакомест во внутренний буфер. Форматирует его в соответствии с флагами `flags`. Затем выводит содержимое буфера в цепочку модулей. Замечание: знак `.` (точка) не занимает отдельного знакоместа при выводе, а выводится в предыдущем знакоместе.
+Записывает число или текстовую строку, начиная с позиции `position`, шириной `width` знакомест во внутренний буфер. Форматирует его в соответствии с флагами `flags`. Затем выводит содержимое буфера в цепочку модулей. Замечание: в случаях когда перед знаком `.` (точка) есть другой символ она не занимает отдельного знакоместа при выводе, а выводится в предыдущем знакоместе.
```
Comment (AlexErAmp) [inline API_ru.md]:
```suggestion
Записывает число или текстовую строку, начиная с позиции `position`, шириной `width` знакомест во внутренний буфер. Форматирует его в соответствии с флагами `flags`. Затем выводит содержимое буфера в цепочку модулей. Замечание: в случаях когда перед знаком `.` (точка) есть другой символ, она не занимает отдельного знакоместа при выводе, а выводится в предыдущем знакоместе.
```

File: src/segm8.cpp [hunk 2]
```diff
@@ -298,16 +301,25 @@ void SegM8::display(String string, uint8_t position, uint8_t width, uint8_t flag
...
     uint8_t trueWidth = 0; // length of result
-    for (i = 0; true; i++) {
-        if (string.length() <= i) // end of string
-            break;
-        else if (trueWidth == width) // calculation end
-            break;
-        else if (string[i] == '.') {
-            trueWidth--;
+
+    uint8_t prev = 0;
```
Comment (AlexErAmp) [inline src/segm8.cpp]: Don't repeat yourself. Can duplicated code be moved into a separate function?

**Review by AlexErAmp — CHANGES_REQUESTED** [review]

File: src/segm8.cpp (hunk 2 above)
Comment (yurybotov) [reply src/segm8.cpp]: Good idea. Done. Look down.

**Review by AlexErAmp — APPROVED** [review]
LGTM

**Review by nomad605dis — APPROVED** [review]
LGTM

**Review by nkrkv — APPROVED** [review]

**Pull request merged**